narwhals/dataframe.py

```diff
@@ -4154,7 +4143,26 @@ def unique(
             │ 1   ┆ a   ┆ b   │
             └─────┴─────┴─────┘
         """
-        return super().unique(subset, keep=keep, maintain_order=maintain_order)
+        if keep not in {"any", "none"}:
+            msg = (
+                "narwhals.LazyFrame makes no assumptions about row order, so only "
+                f"'any' and 'none' are supported for `keep` in `unique`. Got: {keep}."
+            )
+            raise ValueError(msg)
+        if maintain_order:
+            msg = "`maintain_order=True` is not supported for LazyFrame.unique."
+            raise ValueError(msg)
```

Comment:
i think it's important to raise here, else someone passing `maintain_order=True` would potentially have got wrong results